compiler/noirc_evaluator/src/ssa/ir/function.rs

```diff
@@ -217,6 +218,17 @@ impl Function {
             })
             .sum()
     }
+
+    /// Iterate over the numeric constants in the function.
+    pub fn constants(&self) -> impl Iterator<Item = (&FieldElement, &NumericType)> {
```

Comment:
We store a `constants: HashMap<(FieldElement, NumericType), ValueId>` field in the dfg that is used for optimizing the creation of constants. We could probably just expose fetching this field or at least its keys rather than iterating through all the values.

Reply:
Yeah, I wasn't sure what would the team prefer. Everything in the `DataFlowGraph` is `pub(crate)`, as well as the `dfg` field of `Function`. Instead of starting to make this field or that field public, I thought I'd add a helper function to do exactly what I needed for now, to avoid all arguments: can't expose less than this. 

In general I remember `foo.bar.qux.baz` being an anti pattern as well, so it would at least need to be a read-only `.dfg()` method, etc. 

Open to suggestions if there is consensus over how much to expose.